comp/remote-config/rcclient/rcclient.go

```diff
@@ -114,9 +130,57 @@ func (rc rcClient) Start(agentName string) error {
 
 	rc.client.Start()
 
+	if rc.clientHA != nil {
+		rc.clientHA.SetAgentName(agentName)
+		rc.clientHA.Subscribe(state.ProductAgentFailover, rc.haUpdateCallback)
+		rc.clientHA.Start()
+	}
+
 	return nil
 }
 
+func (rc rcClient) haUpdateCallback(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	for cfgPath, update := range updates {
+		haUpdate, err := parseHighAvailabilityConfig(update.Config)
+		if err != nil {
+			pkglog.Errorf("HA update unmarshal failed: %s", err)
+			applyStateCallback(cfgPath, state.ApplyStatus{
+				State: state.ApplyStateError,
+				Error: err.Error(),
+			})
+			continue
+		}
+
+		// Accept and apply only the first valid update. Disregard any remaining, which will be left in the UNKNOWN state
+		if haUpdate != nil && haUpdate.Failover != nil {
+			pkglog.Infof("Setting `ha.failover: %t` through remote config", *haUpdate.Failover)
+			err = settings.SetRuntimeSetting("ha.failover", *haUpdate.Failover, model.SourceRC)
+			if err != nil {
+				pkglog.Errorf("HA failover update failed: %s", err)
+				applyStateCallback(cfgPath, state.ApplyStatus{
+					State: state.ApplyStateError,
+					Error: err.Error(),
+				})
+			} else {
+				applyStateCallback(cfgPath, state.ApplyStatus{State: state.ApplyStateAcknowledged})
+			}
+		} else {
+			// Config update is nil, so we should unset the failover value if it was set via RC previously
+			shouldLog := false
+			if config.Datadog.GetSource("ha.failover") == model.SourceRC {
+				shouldLog = true
+			}
+			config.Datadog.UnsetForSource("ha.failover", model.SourceRC)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 Nitpick:** ‏This is a bit compute heavy (it changes one config Viper instance & then merges all of them again), maybe we should only do this if the source is RC? NBD though.

Reply:
I think it may be worth keeping; the only way we anticipate being able to get here is if `ha.failover` was previously set for the RC source. I don't think it's possible for the _current_ source to _not_ be RC when we get to this point, but if it were, we wouldn't want the unsetting of another source to cause us to defer to a stale RC value, right?

Reply:
> we wouldn't want the unsetting of another source to cause us to defer to a stale RC value, right?

Unsetting a parameter for a source should fall back to the next source. Here only CLI > RC, so unsetting CLI making the agent fall back on RC makes sense to me. 

TBH this is not a big issue as the callback is only called when there's an update to the product, which should only happen when we need to do something anyway.